src/routes/authentication/Authentication.component.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from "react";
+import FormWrapper from "./components/FormWrapper.component";
+import { useLocation, useNavigate } from "react-router-dom";
+import { Container } from "../../shared/ components/Container";
+
+export enum AuthEnum {
+  SIGN_IN = "sign-in",
+  SIGN_UP = "sign-up",
+}
+
+export type FormField = {
+  name: string;
+  label: string;
+  type: string;
+  sm?: number;
+  xs?: number;
+};
+
+const SIGN_UP_FORM_FIELDS = [
+  { name: "firstName", label: "First Name", type: "text", sm: 6, xs: 12 },
+  { name: "lastName", label: "Last Name", type: "text", sm: 6, xs: 12 },
+  { name: "email", label: "Email", type: "email", xs: 12 },
+  { name: "postCode", label: "Post Code", type: "text", xs: 12 },
+  { name: "password", label: "Password", type: "password", xs: 12 },
+];
+
+const SIGN_IN_FORM_FIELDS = [
+  { name: "email", label: "Email", type: "email", xs: 12 },
+  { name: "password", label: "Password", type: "password", xs: 12 },
+];
+
+const Authentication = () => {
+  const [formFields, setFormFields] = useState({});
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  let authType: AuthEnum = location.pathname.includes("sign-in")
+    ? AuthEnum.SIGN_IN
+    : AuthEnum.SIGN_UP;
+
+  const handleFormFieldChange = (
+    e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ) => {
+    const { name, value } = e.target;
+    setFormFields({ ...formFields, [name]: value });
+  };
```

Comment:
Does handleFormFieldChange need to live outside the FormWrapper or will it always be the same?